snuba/consumers/consumer_builder.py

```diff
@@ -48,6 +48,9 @@ def __init__(
         queued_max_messages_kbytes: int,
         queued_min_messages: int,
         strategy_factory_type: StrategyFactoryType,
+        processes: Optional[int],
```

Comment:
From your comment on the whole PR in response to this one:

> I'm looking into the builder changes, if that is one of them, to try and find a solution that I think improves the design. Personally, I'm not really sure how valuable it is to improve that right now, considering it's bookended with the command line entrypoint which suffers the same parameterization issue, as well as the StreamingConsumerStrategyFactory which has the same problem (also mentioned in #1269 (comment).) Realistically, we'll still need runtime checks at some level(s) to ensure that the configuration is valid as long as the configuration is driven by command line arguments that can be passed independently.

Not blocking for what concerns this PR.
I do not really agree with the fact that it is not worth following a builder pattern (at least I mean that for improving the design) because the same issue is present in the CLI. The way we build a consumer internally does not need to be coupled to the way it is built in the CLI and this builder can be used in contexts other than the CLI. 
Then if we can improve the CLI as well, that better. If we stopped relying on CLI and started using config for these parameters even better, at that point we would have a different design and the builder may not be needed.
It is true that as long as we have a CLI we still have the runtime checks somewhere. But since the CLI imposes this need they can be restricted there. Everything else the CLI depends onto (the builder) can still prevent bad configuration statically so those checks are only restricted to the CLI.